tests/integration_tests/performance/test_network_tcp_throughput.py

```diff
@@ -0,0 +1,396 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests the network throughput overhead added by Firecracker."""
+
+import itertools
+import json
+import logging
+import time
+import concurrent.futures
+import re
+import pytest
+from conftest import _test_images_s3_bucket
+from framework.artifacts import ArtifactCollection, ArtifactSet
+from framework.matrix import TestMatrix, TestContext
+from framework.builder import MicrovmBuilder
+from framework.statistics import core, consumer, producer, criteria, \
+    types, function
+from framework.utils import CpuMap, ProcessCpuAffinity, CmdBuilder, run_cmd, \
+    eager_map
+import host_tools.network as net_tools
+
+CONFIG = {
+    "time": 2,  # seconds
+    "omit": 1,
+    "modes": {
+        "g2h": [""],
+        "h2g": ["-R"],
+        "bd": ["", "-R"]
+    },
+    "protocols": [
+        {
+            "name": "tcp",
+            "omit": 1,  # seconds
+            # Used as KBytes - congestion window.
+            "window_size": [10, 64, 128, 256],
+            "pkt_size": [128, 256, 512, 1024],
+        }
+    ],
+    "baseline_bw": {  # Mibps
```

Comment:
this is hard to read... I also suggest to put it in a different file so it can be generated and name them goldvalues, or something more suggestive.
Is pretty hard to understand and change them. I recommend you add a mode where you can generate the gold for example a command like ./test_throughput --generate_goldvalues --delta=0.1 ... which would generate the files.

Reply:
From my experience of running this test, generating gold values is a process which should run the test for many iterations, extracting the minimum and the maximum for each test combination. The gold values contains a target and a delta. The target can be the mean: (MIN + MAX) / 2 and delta: (MAX - MIN) / 2 + a not significant buffer space.

This means that we can not simply overwrite the baselines, running the test once. We do not have an assurance that this process will provide the best resilience to intermittent failures.

I've improved the information presented in the baseline. Please let me know if you still find the baselines hard to read.